### PR TITLE
Copy Paste Tonnage Values

### PR DESCRIPTION
**Pull request by NateHogan**

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            <div class="govuk-grid-column-one-half">
+                <h2 class="govuk-heading-s" id="received-entry">Total received (tonnes)</h2>
+                <textarea class="govuk-textarea govuk-!-width-three-quarters" placeholder="Paste values here" id="ReceievedPastedValues" name="ReceievedPastedValues" style="height: 400px;" aria-labelledby="received-entry"></textarea>
+            </div>
+            <div class="govuk-grid-column-one-half">
+                <h2 class="govuk-heading-s" id="reused-entry">Reused as whole appliances (tonnes)</h2>
+                <textarea class="govuk-textarea govuk-!-width-three-quarters" placeholder="Paste values here" id="ReusedPastedValues" name="ReusedPastedValues" style="height: 400px;" aria-labelledby="reused-entry"></textarea>
+            </div>
+        </div>
+        <button type="submit" class="govuk-button" id="aatf-task-list" data-module="govuk-button" data-prevent-double-click="true">
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml:43]: can you use @Html.Gds() control for this button?

File: src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        @Html.HiddenFor(m => m.Action)
+
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-one-half">
+                <h2 class="govuk-heading-s" id="received-entry">Total received (tonnes)</h2>
+                <textarea class="govuk-textarea govuk-!-width-three-quarters" placeholder="Paste values here" id="ReceievedPastedValues" name="ReceievedPastedValues" style="height: 400px;" aria-labelledby="received-entry"></textarea>
+            </div>
+            <div class="govuk-grid-column-one-half">
+                <h2 class="govuk-heading-s" id="reused-entry">Reused as whole appliances (tonnes)</h2>
+                <textarea class="govuk-textarea govuk-!-width-three-quarters" placeholder="Paste values here" id="ReusedPastedValues" name="ReusedPastedValues" style="height: 400px;" aria-labelledby="reused-entry"></textarea>
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml:40]: check if you can reuse @Html.Gds() control for the textarea too?

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml [hunk 3]
```diff
@@ -85,82 +85,91 @@
...
+                            <p class="govuk-body"><strong>Total</strong></p>
                         </td>
-                        <td class="govuk-table__cell">
-                            <div class="govuk-form-group @Html.Gds().FormGroupClass(m => m.CategoryValues[i], m => m.CategoryValues[i].Received)">
-                                <span class="govuk-visually-hidden" id="category-received-@category.CategoryId">Total received in tonnes for @category.CategoryDisplay</span>
-                                @Html.Gds().ValidationMessageFor(m => m.CategoryValues[i].Received)
-                                @Html.TextBoxFor(m => m.CategoryValues[i].Received, new { @class = "govuk-input tonnage-entry", aria_labelledby = "category-received-" + @category.CategoryId, autocomplete = "off", maxlength = CommonMaxFieldLengths.Tonnage })
-                            </div>
+                        <td class="govuk-table__cell govuk-table__cell--numeric">
+                            <p class="govuk-body  weee-tonnage-right-aligned-display" id="ReceivedTotal">@Model.ReceivedTotal</p>
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml:153]: extra space in front of wee

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.Web/Areas/Aatf/Controllers/ManageEvidenceNotesController.cs [hunk 4]
```diff
@@ -120,6 +126,11 @@ public async Task<ActionResult> CreateEvidenceNote(EditEvidenceNoteViewModel vie
         {
             using (var client = apiClient())
             {
+                if (viewModel != null && viewModel.Action == ActionEnum.CopyAndPaste)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Aatf/Controllers/ManageEvidenceNotesController.cs:129]: probably wouldnt bother doing the null check for the viewmodel here... if thats null we have bigger problems to worry about, same for the edit

File: src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml (hunk 2 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml:40]: There isnt a general GDS one but you could probably use something similar to use Html.TextAreaFor(m => m.Reason, htmlAttributes: new { @class = "govuk-textarea", @maxlength = "200", @title = "You can type up to 200 characters" })....

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml [hunk 5]
```diff
@@ -85,82 +85,91 @@
...
+            </div>
+            <div class="govuk-grid-column-one-half govuk-body govuk-!-font-size-19 govuk-!-padding-top-5" style="padding-left: 1em">
+                <button aria-hidden="true" class="link-submit govuk-link govuk-link--no-visited-state" type="submit" name="Action" value="@ActionEnum.CopyAndPaste" tabindex="-1">Use copy and paste to populate the table</button>
+            </div>
             <label for="CategoryValues" class="govuk-visually-hidden">hidden link for navigation to tonnage error</label> <input id="CategoryValues" class="govuk-visually-hidden" />
             @Html.Gds().ValidationMessageFor(m => m.CategoryValues)
         </div>
 
+            @*<div class="govuk-table govuk-!-margin-bottom-0">
+            <div class="govuk-grid-column-one-half">&nbsp;</div>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml:103]: remove the commented out code

File: src/EA.Weee.Web/Areas/Aatf/Controllers/ManageEvidenceNotesController.cs [hunk 6]
```diff
@@ -120,6 +126,11 @@ public async Task<ActionResult> CreateEvidenceNote(EditEvidenceNoteViewModel vie
...
+                {
+                    sessionService.SetTransferSessionObject(Session, viewModel, SessionKeyConstant.EditEvidenceViewModelKey);
+                    return RedirectToAction("Index", "EvidenceTonnageValueCopyPaste", new { organisationId, returnAction = EvidenceCopyPasteActionConstants.CreateEvidenceNoteAction });
```
[the remark this answers is not in the document]
Comment (NateHogan) [reply src/EA.Weee.Web/Areas/Aatf/Controllers/ManageEvidenceNotesController.cs]: Done

File: src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml (hunk 1 above)
Comment (NateHogan) [reply src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml]: Done

File: src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml (hunk 2 above)
Comment (NateHogan) [reply src/EA.Weee.Web/Areas/Aatf/Views/EvidenceTonnageValueCopyPaste/Index.cshtml]: Changed over, as well as label

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml (hunk 3 above)
Comment (NateHogan) [reply src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml]: Couldn't find extra space

File: src/EA.Weee.Web/Areas/Aatf/Controllers/ManageEvidenceNotesController.cs (hunk 4 above)
Comment (NateHogan) [reply src/EA.Weee.Web/Areas/Aatf/Controllers/ManageEvidenceNotesController.cs]: Null checks removed, force of habit

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml (hunk 5 above)
Comment (NateHogan) [reply src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/_EditEvidenceNotePartial.cshtml]: Removed

**Pull request merged**